README.md

```diff
@@ -1,12 +1,294 @@
+
+​	**cluster_name:** Name of the cluster 
+
+​	**os_credentials:** 
+
+​		**os_admin_username:** Username for the OpenSearch for connecting. This can be set to empty if the security is disable in OpenSearch
+
+​		**os_admin_password:** Password for the OpenSearch for connecting. This can be set to empty if the security is disable in OpenSearch
+
+​	**os_user:** SSH login username
+
+​	**os_group:** 
+
+​	**os_version:** OpenSearch version which needs to be used
+
+​	**os_home: **Default OpenSearch user info
+
+​	**domain_name:** Configure hostnames for OpenSearch nodes which is required to configure SSL
+
+​	**cloud_type:** Name of the cloud infrastructure
+
+​	 **cloud_credentials:**
+
+​		**secret_key:** Secret key for cluster
+
+​		**access_key:** Access key for cluster
+
+​		**pem_file_path:** Path where the pem file is located 
+
+​		**region:** 
+
+​	 **base_node_type:** t2x.large
+
+​	 **number_cpus_per_node:** Total number of CPU present per node
+
+​	 **ram_per_node_in_gb:** Size of RAM used per node (GB)
+
+​	**disk_per_node_in_gb:** Size of DISK used per node (GB)
+
+​	**max_nodes_allowed:** Maximum number of nodes allowed for the cluster
+
+​	**min_nodes_allowed:** Minimum number of nodes allowed for the cluster
+
+**task_details:** Field that contains details on what task should be performed i.e scale_up_by_1 or scale_down_by_1
+
+- **task_name:** Task name indicates the name of the task to recommend by the recommendation engine.
+  **operator:** Operator indicates the logical operation needs to be performed while executing the rules
+  **rules:** Rules indicates list of rules to evaluate the criteria for the recommendation engine.
+
+  - **metric:** Metric indicates the name of the metric. These can be CpuUtil, MemUtil, ShardUtil, DiskUtil
+    **limit: **Limit indicates the threshold value for a metric.
+    **stat:** Stat indicates the statistics on which the evaluation of the rule will happen. These can be AVG, COUNT
+    **decision_period:** Decision Period indicates the time in minutes for which a rule is evaluated.
+
+  
+
+### Sample config.yaml
+
+------
+
+[opensearch-scaling-manager/config.yaml at release_v0.1_dev · maplelabs/opensearch-scaling-manager (github.com)](https://github.com/maplelabs/opensearch-scaling-manager/blob/release_v0.1_dev/config.yaml)
+
+
+
+### Scaling Manager Pre-Requisites
+
+------
+
+- Cluster with OpenSearch installed 
+- OpenSearch version - 1.2.4 and above 
+- Cluster credentials (Username, Password) to access the OpenSearch
+- Cloud credential  (Username, Password) 
+- In AWS we can create a instance by templates which is provided by Domain that is used
+- Launch Template - AWS launch template to spin a new node which has the necessary tags
+- Security certificate to have regex in it to accept the new node 
+- PEM file 
+- SSH aspect -  Port which is used should have permission to connect SSH from one OpenSearch node to another node by which we can install OpenSearch on new node 
```

Comment:
If cloud type is AWS then Security group is configured in such a way that newly spin up node should be reached via ssh.